test/gtest/common/mem_buffer.cc

```diff
@@ -229,11 +229,27 @@ uint64_t mem_buffer::pat(uint64_t prev) {
 }
 
 mem_buffer::mem_buffer(size_t size, ucs_memory_type_t mem_type) :
-    m_mem_type(mem_type), m_ptr(allocate(size, mem_type)), m_size(size) {
+    m_mem_type(mem_type), m_ptr(NULL), m_size(size), m_allocated(false)
+{
+    m_ptr       = allocate(size, mem_type);
+    m_allocated = (m_ptr != NULL);
+}
+
+mem_buffer::mem_buffer(void *ptr, size_t size, ucs_memory_type_t mem_type) :
+    m_mem_type(mem_type), m_ptr(NULL), m_size(size), m_allocated(false)
+{
+    if (ptr == NULL) {
+        m_ptr       = allocate(size, mem_type);
+        m_allocated = (m_ptr != NULL);
```

Comment:
remove m_allocated, check m_ptr!=NULL instead

Reply:
will not work: m_allocated is used to free m_ptr buffer in case if it was allocated in constructor. m_ptr is never NULL because in case if it was NULL - buffer is allocated

Reply:
i see. But allocate() would never return ptr!=NULL. so we can always set m_allocate to false in this ctor, and true in other ctor